packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts

```diff
@@ -534,17 +549,35 @@ export class PrismaSchemaParserService {
     const schema = builder.getSchema();
     const models = schema.list.filter((item) => item.type === MODEL_TYPE_NAME);
 
-    Object.entries(mapper.modelNames).map(([oldName, { newName }]) => {
+    Object.entries(mapper.modelNames).map(([originalName, { newName }]) => {
       models.map((model: Model) => {
+        const originalModelName = findOriginalModelName(mapper, model.name);
+        const originalFieldName = findOriginalFieldName(mapper, model.name);
         const fields = model.properties.filter(
           (property) => property.type === FIELD_TYPE_NAME
         ) as Field[];
         fields.map((field: Field) => {
-          if (field.fieldType === oldName) {
-            mapper.fieldTypes[field.fieldType] = {
-              oldName: field.fieldType,
-              newName,
-            };
+          if (field.fieldType === originalName) {
+            if (
+              mapper.fieldTypes[originalModelName] &&
+              mapper.fieldTypes[originalModelName][originalFieldName]
+            ) {
+              mapper.fieldTypes[originalModelName][originalFieldName][
+                field.fieldType
+              ] = {
+                originalName: field.fieldType,
+                newName,
+              };
+            } else {
+              mapper.fieldTypes[originalModelName] = {
+                [originalFieldName]: {
+                  [field.fieldType]: {
+                    originalName: field.fieldType,
+                    newName,
+                  },
+                },
+              };
```

Comment:
What if ` mapper.fieldTypes[originalModelName] ` exists but ` mapper.fieldTypes[originalModelName][originalFieldName]` doesn't? 
The value `mapper.fieldTypes[originalModelName` will get corrupted and a simple unit test will uncover this.

Try
 ```suggestion
            if (field.fieldType === originalName) {
                mapper.fieldTypes = {
                    ...mapper.fieldTypes,
                    [originalModelName]: {
                        ...mapper.fieldTypes[originalModelName],
                        [originalFieldName]: {
                            ...mapper.fieldTypes[originalModelName][originalFieldName],
                            [field.fieldType]: {
                                originalName: field.fieldType,
                                newName,
                            },
                        },
                    },
                };
            }
```